src/main/java/com/mongodb/hibernate/jdbc/ResultSetAdapter.java

```diff
@@ -0,0 +1,1007 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.jdbc;
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.net.URL;
+import java.sql.Array;
+import java.sql.Blob;
+import java.sql.Clob;
+import java.sql.Date;
+import java.sql.NClob;
+import java.sql.Ref;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.RowId;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.sql.SQLWarning;
+import java.sql.SQLXML;
+import java.sql.Statement;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.Calendar;
+import java.util.Map;
+import org.jspecify.annotations.Nullable;
+
+/**
+ * A {@link java.sql.ResultSet} adapter interface that throws exceptions for all its API methods.
+ *
+ * @see MongoResultSet
+ */
+interface ResultSetAdapter extends ResultSet {
+    @Override
+    default boolean next() throws SQLException {
+        throw new SQLFeatureNotSupportedException("next not implemented");
+    }
+
+    @Override
+    default void close() throws SQLException {
```

Comment:
Is there a clean way to determine which methods are actually not implemented? If not, then I think we should remove method defaults from this class as they are implemented.

Reply:
Would you agree that the methods are "dead code"?

Of course, it requires effort to remove dead code, so it is fine to come back to this sort of thing once most of the changes stop. If you think we should take that approach instead, how about a Jira ticket?

Reply:
I found it is easy to narrow down the "dead code" in IDE (there is "1 override" hints), so I removed them in this PR. Will do the same for other JDBC "dead code" in later PR.